File: Domasna_2/prototip/stocktopus-backend/src/main/java/mk/finki/ukim/mk/stocktopusbackend/model/dto/StockIndicatorsDTO.java
package mk.finki.ukim.mk.stocktopusbackend.model.dto;

import java.util.Date;

public record StockIndicatorsDTO(Long id,
                                 Long stockId,
                                 Date date,
                                 String timeframe,
                                 Double sma50,
                                 Double ema50,
                                 Double rsi,
                                 Double macd,
                                 Double signalValue,
                                 String decision) {
}

File: Domasna_2/prototip/stocktopus-frontend/src/model/dto/stockIndicatorsDTO.ts
export interface StockIndicatorsDTO {
    id: number;
    stockId: number;
    date: Date;
    timeframe: string;
    sma50: number;
    sma200: number;
    ema50: number;
    ema200: number;
    rsi: number;
    macd: number;
    stochasticOscillator: number;
    cci: number;
    williamsR: number;
    bollingerHigh: number;
    bollingerLow: number;
    signal: string;
}